src/bin/tripadvisor2mimir.rs

```diff
@@ -145,16 +145,18 @@ async fn load_and_index_tripadvisor(settings: Settings) {
             .filter(|(ta_id, _)| future::ready(indexed_documents.contains(ta_id)))
             .flat_map(|(ta_id, reviews)| {
                 count_ok += 1;
-                futures::stream::iter(reviews.into_iter().enumerate().map(
-                    move |(review_id, review)| {
+                let update_operations = reviews
+                    .into_iter()
+                    .enumerate()
+                    .map(|(review_id, review)| {
                         let ident = format!("properties.ta:reviews:{review_id}");
-                        let op = UpdateOperation::Set {
+                        UpdateOperation::Set {
                             ident,
                             value: review,
-                        };
-                        (build_id(ta_id), op)
-                    },
-                ))
+                        }
+                    })
+                    .collect();
+                futures::stream::iter([(build_id(ta_id), update_operations)])
```

Comment:
Here we are flattening a stream that always contains exactly 1 element, so I think we could just use `.map` instead of `.flat_map`

```suggestion
                (build_id(ta_id), update_operations)
```